server/handlers/project_command_output_handler.go

```diff
@@ -51,8 +57,13 @@ func (p *DefaultProjectCommandOutputHandler) Send(ctx models.ProjectCommandConte
 	}
 }
 
-func (p *DefaultProjectCommandOutputHandler) Receive(projectInfo string, callback func(msg string) error) error {
-	ch := p.addChan(projectInfo)
+func (p *DefaultProjectCommandOutputHandler) Receive(projectInfo string, ch chan string, callback func(msg string) error) error {
+
+	// Avoid deadlock when projectOutputBuffer size is greater than the channel (currently set to 1000)
+	// Running this as a goroutine allows for the channel to be read in callback
+	go p.addChan(ch, projectInfo)
```

Comment:
note: I think my suggestion would eliminate the need for this. Kinda weird to be creating a new thread for every ws conn just to register it.

Reply:
Yup, will rethink about how we do this in a later PR.